include/mqtt/endpoint.hpp

```diff
@@ -7758,7 +7758,10 @@ class endpoint : public std::enable_shared_from_this<endpoint<Socket, Mutex, Loc
         if (connected_) {
             connected_ = false;
             mqtt_connected_ = false;
-            shutdown_from_server(*socket_);
+            {
+                boost::system::error_code ec;
```

Comment:
Is there any meaningful check that can be made on the error_code? 

I can't think of any. Some kind of log message perhaps?

Reply:
I want to avoid throwing exception. I think that it should be logged as **warning** level in the future.